src/main/java/wolox/training/models/Book.java

```diff
@@ -55,4 +56,48 @@
     @ManyToMany(mappedBy = "books")
     private List<User> users;
 
+    public void setGenre(String genre) {
+        Preconditions.checkNotNull(genre,"Please check the genre supplied, its null!");
```

Comment:
- [ ] This message should be a constant in another class, and here use the `String.format` method. This is how **DRY** is applied.
- [ ]  In addition, you should validate that the field is not empty(apply for every not null field).